src/accel/Logger.cc

```diff
@@ -110,7 +111,11 @@ void MtLogger::operator()(Provenance prov, LogLevel lev, std::string msg)
 Logger make_mt_logger(G4RunManager const& runman)
 {
     return Logger(MpiCommunicator{},
+#if G4VERSION_NUMBER < 1070
+                  MtLogger{2},
+#else
```

Comment:
As discussed in person, let's see if we can get the number of threads by dynamic casting

Reply:
Not sure how you usually handle exceptions in Celeritas. I went with the `CELERITAS_TRY_HANDLE` macro

Reply:
Yeah, I would've just done a `auto* mt_rm = dynamic_cast<Manager const*>(&runman)`. I think you should move the code from the `HitRootIO.cc` into here as a free function `int GetNumberOfThreads(G4MTRunManager const&)`, which you can reuse.

Reply:
Good point. Since `celeritas/ext/GeantSetup.hh` creates and uses a run manager, how about there? If that's good enough, let's call the function `get_num_threads` since that directory is inside celeritas (not meant to be Geant4-user-facing).